Map/Map.cpp

```diff
@@ -0,0 +1,358 @@
+//--------------------------------Continent class----------------------------//
+//-------------- Constructors --------------//
+Continent::Continent(string continentName){
+    this->continentName = continentName;
+}
+
+Continent::Continent(const Continent& original){ //copy constructor
+    continentName = original.continentName;
+}
+
+//-------------- Overloads --------------//
+//overload assignment operator
+Continent& Continent::operator=(const Continent& rhs){
+    if(this != &rhs){
+        continentName = rhs.continentName;
+    }
+    return *this;
+}
+
+//-------------- Getters --------------//
+string Continent::getContinentName(){
+    return continentName;
+}
+
+
+//-------------- Setters --------------//
+void Continent::setContinentName(string newContinentName){
+    continentName = newContinentName;
+}
+
+
+//--------------------------------Territory class----------------------------//
+//-------------- Constructors --------------//
+Territory::Territory(){}
+
+Territory::Territory(string territoryName, Continent* continent, int ownerId, int numberOfArmies){
+    this->territoryName = territoryName;
+    this->ownerId = ownerId;
+    this->numberOfArmies = numberOfArmies;
+    this->continent = continent;
+}
+
+Territory::Territory(string territoryName, Continent* continent){
+    this->territoryName = territoryName;
+    this->continent = continent;
+}
+
+Territory::Territory(const Territory& original){//copy constructor
+    territoryName = original.territoryName;
+    ownerId = original.ownerId;
+    numberOfArmies = original.numberOfArmies;
+    continent = original.continent;//we want shallow copy
+} 
+
+//-------------- Overloads --------------//
+//overload assignment operator
+Territory& Territory::operator=(const Territory& rhs){
+        if(this != &rhs){
+            territoryName = rhs.territoryName;
+            ownerId = rhs.ownerId;
+            numberOfArmies = rhs.numberOfArmies;
+            continent = rhs.continent; //shallow is wanted
+    }
+    return *this;
+}
+
+//-------------- Getters --------------//
+string Territory::getTerritoryName(){
+    return territoryName;
+}
+
+int Territory::getOwnerId(){
+    return ownerId;
+}
+
+int Territory::getNumberOfArmies(){
+    return numberOfArmies;
+}
+
+Continent* Territory::getContinent(){
+    return continent;
+}
+
+//-------------- Setters --------------//
+void Territory::setTerritoryName(string newTerritoryName){
+    territoryName = newTerritoryName;
+}
+void Territory::setOwnerId(int newOwnerId){
+    ownerId = newOwnerId;
+}
+    
+void Territory::setNumberOfArmies(int newNumberOfArmies){
+    numberOfArmies = newNumberOfArmies;
+}
+
+void Territory::setContinent(Continent* newContinent){
+    this->continent = newContinent;
+}
+
+//-------------------------- Node class  --------------------------//
+Node::Node(){}
+
+Node::Node(Territory data){
+    Territory* temp = new Territory(data.getTerritoryName(), data.getContinent());
+    this->data = temp;
```

Comment:
Why is it in two steps? I think both lines can be combined into one
this->data = new Territory(data.getTerritoryName(), data.getContinent());

Reply:
You are right. I've updated it accordingly.